File: CS 1450 Object Oriented Programming/Finals/Code/finals_5.py
class Building():

	def __init__(self, name, size = "Medium"):

		self.name = name
		self.hours = [0,0]
		if size == "Medium":
			self.size = "Medium"
		else:
			self.size = size

	def set_open_hours(self, open_time, close_time):
		self.hours[0] = open_time
		self.hours[1] = close_time
		print("The buildings open time has been updated, now the building will open at:", 
			self.hours[0], "and will close at", self.hours[1], "\n")

	def set_name(self, new_name):
		self.name = new_name
		print("The building has been renamed to:", self.name, "\n")

	def get_name(self):
		return self.name

	def get_size(self):
		return self.size

	def get_hours(self):
		return self.hours

class Storage(Building):

	def __init__(self, name, size = "Big"):
		self.name = name
		self.size = size
		self.hours = [0,0]

class Corner_Store(Building):

	def __init__(self, name, size = "Small"):
		self.name = name
		self.size = size
		self.hours = [0,0]



building1 = Building("Walmart")
building2 = Storage("Amazon Storage")
building3 = Corner_Store("Bob's Deli")

building1.set_open_hours(9, 11)
building2.set_open_hours(1, 10)
building3.set_open_hours(8, 3)


buildings_list = [building1, building2, building3]

for building in buildings_list:
	print("The buildings name is:", building.get_name())
	print("The buildings open hours are:", building.get_hours())
	print("The buildings size is:", building.get_size())
	print("==================")

File: CS 3140 The C++ Language/Finals/Code/finals_5.cpp
#include <iostream>
#include <string>
using namespace std;

class Building
{
protected:
	string name;
	string size;
	int closed, opened;
public:
	Building(string building_name)
	{
		name = building_name;
		size = "Medium";
		opened = 9;
		closed = 5;
	}

	void set_open_hours(int hour)
	{
		opened = hour;
		cout << "The building's opening hour was updated. Now the building opens at: " << opened << endl;
	}

	void set_closed_hours(int hour)
	{
		closed = hour;
		cout << "The building's closing hour was updated. Now the building closes at: " << closed << endl;
	}

	void set_name(string new_name)
	{
		name = new_name;
	}

	string get_name()
	{
		return name;
	}

	string get_size()
	{
		return size;
	}

	int get_open_hours()
	{
		return opened;
	}

	int get_closed_hours()
	{
		return closed;
	}

	void show_info()
	{
		cout << "The buildings name is: " << name << endl;
		cout << "The buildings open hour is: " << opened << endl;
		cout << "The buildings closed hour is: " << closed << endl;
		cout << "The buildings size is: " << size << endl;
		cout << "=======================================" << endl;
	}

};

class Storage:public Building
{
public:

	Storage(string building_name):Building(building_name)
	{
		size = "Big";
		opened = 1;
		closed = 1;
	}
};

class Corner_Store:public Building
{
public:

	Corner_Store(string building_name):Building(building_name)
	{
		size = "Small";
		opened = 8;
		closed = 3;
	}
};

int main()
{
	Building building1("Walmart");
	Storage building2("Amazon Warehouse");
	Corner_Store building3("Abraham's Deli");

	building1.show_info();
	building2.show_info();
	building3.show_info();

	return 0;
}